src/storage/v3/request_helper.cpp

```diff
@@ -351,10 +352,15 @@ EdgeFiller InitializeEdgeFillerFunction(const msgs::ExpandOneRequest &req) {
 }
 
 bool FilterOnVertex(DbAccessor &dba, const storage::v3::VertexAccessor &v_acc, const std::vector<std::string> &filters,
-                    const std::string_view node_name) {
-  return std::ranges::all_of(filters, [&node_name, &dba, &v_acc](const auto &filter_expr) {
-    auto res = ComputeExpression(dba, v_acc, std::nullopt, filter_expr, node_name, "");
-    return res.IsBool() && res.ValueBool();
+                    const std::string_view node_name, const std::optional<EdgeAccessor> &e_acc) {
+  return std::ranges::all_of(filters, [&node_name, &dba, &v_acc, &e_acc](const auto &filter_expr) {
+    TypedValue result;
+    if (e_acc) {
+      result = ComputeExpression(dba, v_acc, e_acc, filter_expr, "", node_name);
+    } else {
+      result = ComputeExpression(dba, v_acc, e_acc, filter_expr, node_name, "");
+    }
```

Comment:
When you evaluate an expression on an edge, then you have to provide the vertex accessor and vertex name also.